searx/query.py

```diff
@@ -63,100 +70,225 @@ def _parse_query(self):
+            # show some example queries
+            self.autocomplete_list.append(":en")
+            self.autocomplete_list.append(":en_us")
+            self.autocomplete_list.append(":english")
+            self.autocomplete_list.append(":united_kingdom")
+        else:
+            for lc in language_codes:
+                lang_id, lang_name, country, english_name = map(str.lower, lc)
+
+                # check if query starts with language-id
+                if lang_id.startswith(value):
+                    if len(value) <= 2:
+                        self.autocomplete_list.append(':' + lang_id.split('-')[0])
+                    else:
+                        self.autocomplete_list.append(':' + lang_id)
+
+                # check if query starts with language name
+                if lang_name.startswith(value) or english_name.startswith(value):
+                    self.autocomplete_list.append(':' + lang_name)
+
+                # check if query starts with country
+                if country.startswith(value.replace('_', ' ')):
+                    self.autocomplete_list.append(':' + country.replace(' ', '_'))
+
+    def _language_parse(self, value, enable_autocomplete):
+        found = False
+        # check if any language-code is equal with
+        # declared language-codes
+        for lc in language_codes:
+            lang_id, lang_name, country, english_name = map(str.lower, lc)
+
+            # if correct language-code is found
+            # set it as new search-language
+
+            if (value == lang_id
+                or value == lang_name
+                or value == english_name
+                or value.replace('-', ' ') == country)\
+               and value not in self.languages:
+                found = True
+                lang_parts = lang_id.split('-')
+                if len(lang_parts) == 2:
+                    self.languages.append(lang_parts[0] + '-' + lang_parts[1].upper())
+                else:
+                    self.languages.append(lang_id)
+                # to ensure best match (first match is not necessarily the best one)
+                if value == lang_id:
+                    break
+
+        # user may set a valid, yet not selectable language
+        if VALID_LANGUAGE_CODE.match(value):
+            lang_parts = value.split('-')
+            if len(lang_parts) > 1:
+                value = lang_parts[0].lower() + '-' + lang_parts[1].upper()
+            if value not in self.languages:
+                self.languages.append(value)
+                found = True
+
+        if found:
+            return True
+        elif enable_autocomplete:
+            self._language_autocomplete(value)
+            return False
+
+    def _external_bang_autocomplete(self):
+        self.autocomplete_list.append('!!g')
+        self.autocomplete_list.append('!!ddg')
+        self.autocomplete_list.append('!!bing')
+
+    def _external_bang_parse(self, value, enable_autocomplete):
```

Comment:
Please extract `_bang_autocomplete` in a similar fashion as in `_language_parse`.